prettierx: Merge Prettier 2.3.0 updates - draft 1

adalinesimonian opened the pull request:
Closes #548
Closes #414

Many, many hours of work later, this is starting to look good! Still needs work, though.

Merge conflicts have been resolved, moving the appropriate changes in prettierx into their corresponding new locations with the multiple refactors made upstream. ~~However, about 374 out of 1189 test suites are failing - one, for example, because parentheses are not correctly being printed around multi-line conditionals.~~

~~These failing tests shouldn't be that hard to resolve; it seems that in most of the cases, at least from a cursory glance, multiple tests are failing due to the same sets of bugs created from the merge. It will take a bit of effort and time to resolve but then again, so does merging a massive amount of changes into a fork and, well, at least that's mostly done now!~~

~~Tests now pass (at least locally)!~~ ~~Actually, it turns out there are 4 test suites that still fail - I hadn't run them because I was unaware of the `FULL_TEST` flag.~~ Test failures resolved.

I've enabled edits by maintainers so if y'all would like to have a crack at anything, please, be my guest. In the meantime, I'll keep trying to iron this stuff out.

- [x] Make test suites pass
  - [x] 374 failing → 0 failing
- [x] Write any new tests, if necessary
- [x] Make sure that dependency changes in prettierx, such as flow being optional, are still in effect if possible to keep
- [x] Ensure CI/workflows are valid
- [x] Conform changes to lint rules
- [x] Document changes
- [ ] Clean-up

renovate[bot] committed:
prettierx: Update eslint -> 7.22.0 - devDependency (#498)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
prettierx: Update @rollup/plugin-replace -> 2.4.2 - devDependency (#500)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
prettierx: Update execa -> 5.0.0 - devDependency (#485)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
prettierx: Update tempy -> 1.0.1 - devDependency (#491)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

Commit:
prettierx: update @babel/core -> 7.13.15 --dev

Commit:
prettierx: update @babel/preset-env -> 7.13.15 --dev

Commit:
prettierx: update eslint -> 7.24.0 --dev

Commit:
prettierx: update eslint-plugin-react -> 7.23.2 --dev

Commit:
prettierx: update jest-watch-typeahead -> 0.6.2 --dev

Commit:
prettierx: update @babel/code-frame -> 7.12.13 (...)

and update a test snapshot

renovate[bot] committed:
prettierx: Update ci-info -> 3.1.1 - dependency (#480)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
prettierx: Update diff -> 5.0.0 - dependency (#482)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
prettierx: Update globby -> 11.0.3 - dependency (#499)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
prettierx: Update mem -> 8.1.0 - dependency (#486)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
prettierx: Update postcss-less -> 4.0.1 - dependency (#487)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
prettierx: Update semver -> 7.3.5 - dependency (#508)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
prettierx: Update rollup -> 2.45.1 - devDependency (#507)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

Commit:
prettierx: no code coverage in GitHub action

do not enable code coverage in .github/workflows/dev-test.yml

renovate[bot] committed:
prettierx: Update resolve -> 1.20.0 - dependency (#465)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

Commit:
prettierx: remove Prettier website code

(should avoid some security warnings reported by GitHub)

dependabot[bot] committed:
build(deps): bump node-fetch from 2.1.2 to 2.6.1 in /scripts/release (#374)

Bumps [node-fetch](https://github.com/bitinn/node-fetch) from 2.1.2 to 2.6.1.
- [Release notes](https://github.com/bitinn/node-fetch/releases)
- [Changelog](https://github.com/node-fetch/node-fetch/blob/master/docs/CHANGELOG.md)
- [Commits](https://github.com/bitinn/node-fetch/compare/v2.1.2...v2.6.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

renovate[bot] committed:
prettierx: Update tslib -> 2.2.0 - dependency (#515)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
prettierx: Update eslint-plugin-unicorn -> 29.0.0 - devDependency (#510)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

Commit:
prettierx: tslib back to 1.14.1

(seems to remove an extra semver entry from yarn.lock as well)

to avoid an extra dependency warning

effectively reverts 44343c4c49367217d8cd38665dcafdd0de6b2988
(PR #515)

Commit:
prettierx: update dependencies notes in package.json

Commit:
prettierx: update yarn.lock --dev (#519)

* remove jest-watch-typeahead from devDependencies
* remove snapshot-diff from devDependencies
* remove jest from devDependencies
* remove babel-loader from devDependencies
* remove terser-webpack-plugin from devDependencies
* remove webpack from devDependencies
* add jest (version 26.6.3) to devDependencies
* add jest-watch-typeahead (version 0.6.2) to devDependencies
* add snapshot-diff (version 0.8.1) to devDependencies
* add webpack@4.46.0 to devDependencies
* add babel-loader (version 8.2.2) to devDependencies
* add terser-webpack-plugin@4.2.3 to devDependencies

Commit:
prettierx: version 0.18.0

Commit:
prettierx: start 0.18.1-dev

renovate[bot] committed:
prettierx: Update rollup -> 2.45.2 - devDependency (#520)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
prettierx: Update eslint-config-prettier -> 8.2.0 - devDependency (#521)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

thorn0 committed:
Bump Prettier dependency to 2.3.0

thorn0 committed:
Git blame ignore 2.3.0

thorn0 committed:
Update dependents count

thorn0 committed:
Fix release script

thorn0 committed:
Add 2.3 blog post (#10659)

* Start 2.3 blog post

* Apply suggestions from code review

Co-authored-by: Alex Rattray <rattray.alex@gmail.com>

* Regenerate the post

* Regenerate blog post

* Regenerate post

* Edit changelog entries

* Regenerate blog post

Co-authored-by: Alex Rattray <rattray.alex@gmail.com>
Co-authored-by: fisker Cheung <lionkay@gmail.com>

thorn0 committed:
Clean changelog_unlreleased

thorn0 committed:
Fix a link in the blog post

fisker committed:
Remove `.prettierrc` parser override (#10845)

fisker committed:
Use async APIs in CLI (#10841)

* Use async API in CLI

* Remove `synchronous-promise`

fisker committed:
Drop Node.js 10 support for development "build" (#10842)

fisker committed:
Use `Object.fromEntries()`, `Array#flat()` and `Array#flatMap()` (#10853)

fisker committed:
Use fsPromises in CLI (#10852)

fisker committed:
Update babel config (#10855)

fisker committed:
Fix `.eslintignore` (#10875)

thorn0 committed:
Update commands.md

pdavies committed:
Fix #5599, long yaml keys trigger explicit mapping (#10874)

Fix #5599, long yaml keys trigger explicit mapping

fisker committed:
Style: Remove useless flags (#10861)

fisker committed:
Switch some scripts to ESM (#10867)

dependabot[bot] committed:
Build(deps-dev): Bump @babel/preset-env from 7.14.1 to 7.14.2 (#10897)

Bumps [@babel/preset-env](https://github.com/babel/babel/tree/HEAD/packages/babel-preset-env) from 7.14.1 to 7.14.2.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.14.2/packages/babel-preset-env)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump eslint-plugin-import from 2.22.1 to 2.23.2 (#10893)

Bumps [eslint-plugin-import](https://github.com/benmosher/eslint-plugin-import) from 2.22.1 to 2.23.2.
- [Release notes](https://github.com/benmosher/eslint-plugin-import/releases)
- [Changelog](https://github.com/benmosher/eslint-plugin-import/blob/master/CHANGELOG.md)
- [Commits](https://github.com/benmosher/eslint-plugin-import/compare/v2.22.1...v2.23.2)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.36.2 to 5.37.0 (#10891)

Bumps [webpack](https://github.com/webpack/webpack) from 5.36.2 to 5.37.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.36.2...v5.37.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump esm-utils from 1.0.1 to 1.1.0 (#10883)

Bumps [esm-utils](https://github.com/fisker/esm-utils) from 1.0.1 to 1.1.0.
- [Release notes](https://github.com/fisker/esm-utils/releases)
- [Commits](https://github.com/fisker/esm-utils/compare/v1.0.1...v1.1.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump @glimmer/syntax from 0.79.0 to 0.79.1 (#10894)

Bumps [@glimmer/syntax](https://github.com/glimmerjs/glimmer-vm) from 0.79.0 to 0.79.1.
- [Release notes](https://github.com/glimmerjs/glimmer-vm/releases)
- [Changelog](https://github.com/glimmerjs/glimmer-vm/blob/master/CHANGELOG.md)
- [Commits](https://github.com/glimmerjs/glimmer-vm/compare/v0.79.0...v0.79.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump flow-parser from 0.150.1 to 0.151.0 (#10899)

Bumps [flow-parser](https://github.com/facebook/flow) from 0.150.1 to 0.151.0.
- [Release notes](https://github.com/facebook/flow/releases)
- [Changelog](https://github.com/facebook/flow/blob/master/Changelog.md)
- [Commits](https://github.com/facebook/flow/compare/v0.150.1...v0.151.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fisker committed:
Remove dead code (#10869)

fisker committed:
Use `join` in `replaceNewlinesWithLiterallines` (#10860)

fisker committed:
Use fsPromise in `getFileContentOrNull` (#10859)

fisker committed:
Reduce use of `Array#reduce()` (#10863)

Co-authored-by: Georgii Dolzhykov <thorn.mailbox@gmail.com>

dependabot[bot] committed:
Build(deps-dev): Bump @babel/core from 7.14.0 to 7.14.2 (#10896)

dependabot[bot] committed:
Build(deps-dev): Bump eslint-plugin-unicorn from 31.0.0 to 32.0.1 (#10882)

dependabot[bot] committed:
Build(deps-dev): Bump terser-webpack-plugin from 5.1.1 to 5.1.2 (#10900)

dependabot[bot] committed:
Build(deps): Bump find-parent-dir from 0.3.0 to 0.3.1 (#10887)

dependabot[bot] committed:
Build(deps-dev): Bump @babel/types from 7.14.1 to 7.14.2 (#10885)

Bumps [@babel/types](https://github.com/babel/babel/tree/HEAD/packages/babel-types) from 7.14.1 to 7.14.2.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.14.2/packages/babel-types)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump rollup from 2.47.0 to 2.48.0 (#10895)

dependabot[bot] committed:
Build(deps): Bump @angular/compiler from 11.2.13 to 11.2.14 (#10888)

dependabot[bot] committed:
Build(deps-dev): Bump @glimmer/reference from 0.79.0 to 0.79.1 (#10889)

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.36.2 to 5.37.0 in /website (#10884)

Bumps [webpack](https://github.com/webpack/webpack) from 5.36.2 to 5.37.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.36.2...v5.37.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump @typescript-eslint/typescript-estree (#10892)

sosukesuzuki committed:
Add JSDoc type for isNonEmptyArray (#10902)

fisker committed:
Move `replaceEndOfLineWith` to `docUtils` (#10862)

fisker committed:
Style: Use `dotAll` regexp (#10864)

dependabot[bot] committed:
Build(deps): Bump @babel/parser from 7.14.1 to 7.14.3 (#10908)

Bumps [@babel/parser](https://github.com/babel/babel/tree/HEAD/packages/babel-parser) from 7.14.1 to 7.14.3.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.14.3/packages/babel-parser)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

sosukesuzuki committed:
Add heuristics to detect type aliases that has complex typeparameters (#10901)

fisker committed:
Add `__babel_estree` parser (#10910)

fisker committed:
Reduce `parser-graphql` size (#10903)

thorn0 committed:
docs: make "A note on formatting reversibility" linkable

thorn0 committed:
Update CONTRIBUTING.md

thorn0 committed:
Update CONTRIBUTING.md

tkalmar committed:
Allow `:` as class prefix-delimeter (#10906)

Co-authored-by: Georgii Dolzhykov <thorn.mailbox@gmail.com>

dependabot[bot] committed:
Build(deps-dev): Bump eslint from 7.26.0 to 7.27.0 (#10932)

Bumps [eslint](https://github.com/eslint/eslint) from 7.26.0 to 7.27.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.26.0...v7.27.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @babel/core from 7.14.2 to 7.14.3 (#10935)

Bumps [@babel/core](https://github.com/babel/babel/tree/HEAD/packages/babel-core) from 7.14.2 to 7.14.3.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.14.3/packages/babel-core)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump @glimmer/syntax from 0.79.1 to 0.79.2 (#10931)

Bumps [@glimmer/syntax](https://github.com/glimmerjs/glimmer-vm) from 0.79.1 to 0.79.2.
- [Release notes](https://github.com/glimmerjs/glimmer-vm/releases)
- [Changelog](https://github.com/glimmerjs/glimmer-vm/blob/master/CHANGELOG.md)
- [Commits](https://github.com/glimmerjs/glimmer-vm/compare/v0.79.1...v0.79.2)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.37.0 to 5.37.1 in /website (#10930)

Bumps [webpack](https://github.com/webpack/webpack) from 5.37.0 to 5.37.1.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.37.0...v5.37.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump eslint-plugin-import from 2.23.2 to 2.23.3 (#10929)

dependabot[bot] committed:
Build(deps): Bump @typescript-eslint/typescript-estree (#10934)

dependabot[bot] committed:
Build(deps-dev): Bump @glimmer/reference from 0.79.1 to 0.79.2 (#10933)

j-f1 committed:
Fix repositioning the cursor at the end of a range format (#10938)

* Adjust the upper bound on cursorOffset when range formatting

* fix implementation

* Add changelog entry

* Update changelog entry to use PR number

adalinesimonian committed:
Merge tag '2.3.0' into adalinesimonian/merge-prettier-2.3.0

adalinesimonian committed:
fix(tests): PRETTIER_DIR → PRETTIERX_DIR

SevenOutman committed:
Support stylelintrc (#10924)

* Infer JSON/YAML parser for .stylelintrc

* Update test snapshots

* Update src/language-yaml/embed.js

Co-authored-by: fisker Cheung <lionkay@gmail.com>

* Add changelog for stylelintrc support

* Update embed.js

* style: Fix prettier linting problems

Co-authored-by: fisker Cheung <lionkay@gmail.com>

sosukesuzuki committed:
Add heuristics to detect assignments whose id has complex type annotations (#10916)

fisker committed:
Remove more `Array#reduce()` (#10907)

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.37.0 to 5.37.1 (#10936)

sosukesuzuki committed:
refactor: Add isObjectProperty util function (#10942)

* Add isObjectProperty

* Rename to isProperty

* Address review

* Address review2

* Make isProperty strict

* Rename to isObjectProperty

sosukesuzuki committed:
Support TypeScript 4.3 (#10945)

* Update deps

* Add tests

* Add changelog

thorn0 committed:
Fix incorrectly wrapped arrow functions with return types (#10940)

fisker committed:
Update `jest` to v27 (#10947)

fisker committed:
Update build script (#10946)

fisker committed:
Move `flow` node to `flow.js` (#10449)

fisker committed:
Minor tweak to js printer (#10954)

brody2consult commented on package.json:
```suggestion
```

---

@adalinesimonian please ignore this, my apologies.

Commit:
FIX-UP: remove flow-parser & typescript from dependencies in package.json

adalinesimonian committed:
feat: add inherited option support

Inherited option support allows propogating certain options to others.

For example, yamlBracketSpacing inherits bracketSpacing. This means that
when bracketSpacing is set, but yamlBracketSpacing isn't, whatever value
bracketSpacing is set to is inherited by yamlBracketSpacing. However, if
yamlBracketSpacing is set, its value will not be overriden by the value
set to bracketSpacing.

This allows prettierx to honour prettier's more broad settings
losing specificity using prettierx's more granular options. This greatly reduces friction
between upstream and this fork by allowing prettier's tests to pass
without increasing change surface area by having to make changes to most
prettier tests. This feature also makes
prettierx even more of a drop-in replacement for prettier since
prettier's options continue to be honoured.

- Added inherited option support
- bracketSpacing reintroduced
- Granular bracketSpacing replacements now inherit bracketSpacing value
- Added JSDoc type annotations to language options exports

adalinesimonian committed:
chore(support): Bump prettier version to 2.3.0

adalinesimonian committed:
fix: pass options to printTemplateExpressions

adalinesimonian committed:
fix(graphql): bracketSpacing → graphqlCurlySpacing

adalinesimonian committed:
fix(modules): bracketSpacing → importCurlySpacing

sosukesuzuki committed:
Avoid to break after assignment for call expression that has complex type parameters (#10949)

* Avoid break-after-operator when call expr has complex type params

* Add tests

* Add changelog

* Fix issue number

* Update snapshpts

* Address review

* Support Flow

* Rename parameters => arguments

* Remove isCallExpression

sosukesuzuki committed:
Update to TypeScript 4.3 (#10959)

* Update links

* Update changelog_unreleased/typescript/10945.md

Co-authored-by: fisker Cheung <lionkay@gmail.com>

* Install TS 4.3.2

Co-authored-by: fisker Cheung <lionkay@gmail.com>

adalinesimonian committed:
fix(breakBeforeElse): correct logic

adalinesimonian committed:
fix(needs-parens): comment out prettierx changes

These conditions seem to no longer be necessary. Commenting them out in
case they need to be restored.

adalinesimonian committed:
fix(spaceInParens): correct logic

- Only add spaces if none have been added already
- Ensure addedLine is set correctly
- Add startsWithSpace utility func to check for existing spaces in front

adalinesimonian committed:
fix(modules): don't break line when we shouldn't

adalinesimonian committed:
fix(spacing): discern objects, type annotations

use objectCurlySpacing for objects and typeCurlySpacing for type
annotations, aliases, and declarations

adalinesimonian committed:
fix(properties): pass node instead of path

adalinesimonian committed:
refactor(tests): relocate flow, ts standard tests

In accordance with upstream test location refactor.

adalinesimonian committed:
fix(tests): move jsx spacing test to correct path

adalinesimonian committed:
fix(tests): fix syntax errors blocking tests

Errors are now thrown for certain errors picked up by parsers that
weren't before. Correcting syntax for those which should be fixed and
setting expected errors for those that should be expected to error out.

adalinesimonian committed:
fix(tests/yaml): ensure parser throws error

adalinesimonian committed:
chore(editorcfg): keep eol whitespace in tests

Keep editor from removing trailing whitespace in test files.

adalinesimonian committed:
fix(tests/integration): add missing properties

adalinesimonian committed:
fix(tests/integration): update snapshots

adalinesimonian committed:
fix(tests/html): expect vue parser to throw error

adalinesimonian committed:
test(format): test bracketSpacing option

adalinesimonian committed:
test(js/function-paren): typeBracketSpacing: false

adalinesimonian committed:
test(js\empty-paren-cmnt): remove duplicate test

adalinesimonian committed:
test(fmt/bin-expr/like-regex): expect espree error

adalinesimonian committed:
test(fmt/js/empty-paren-cmnt): expect espree error

adalinesimonian committed:
test(fmt/js/cmnt-closure-tc): only test w/ babel

adalinesimonian committed:
test: update snapshots

adalinesimonian committed:
fix(pkg): bin-prettier.js → bin-prettierx.js

adalinesimonian committed:
chore(lint): fix unknown word lint error

adalinesimonian committed:
fix(lint): correct env/parserOptions for website

adalinesimonian committed:
fix(lint): conform to lint rules

adalinesimonian committed:
fix(test/standalone): prettier → prettierx

adalinesimonian committed:
fix(test/standalone): prettier → prettierx

adalinesimonian committed:
fix(lint/tests-standalone): fix long line

adalinesimonian committed:
docs(options): document bracketSpacing inheritance

adalinesimonian committed:
chore(pkg): use build script

Doesn't look like the build script verbatim from upstream causes any
issues in this version

adalinesimonian committed:
[prettierx] docs(CONTRIBUTING): prettier → prettierX etc. (#553)

Co-authored-by: Adaline Valentina Simonian <adalinesimonian@gmail.com>
Co-authored-by: Christopher J. Brody <chris.brody+brodybits@gmail.com>

adalinesimonian committed:
test: add option inheritance test

adalinesimonian committed:
Merge branch 'dev' of https://github.com/brodybits/prettierx into adalinesimonian/merge-prettier-2.3.0

adalinesimonian committed:
test(config): update snapshots

dependabot[bot] committed:
Build(deps): Bump jest-docblock from 26.0.0 to 27.0.1 (#10985)

Bumps [jest-docblock](https://github.com/facebook/jest/tree/HEAD/packages/jest-docblock) from 26.0.0 to 27.0.1.
- [Release notes](https://github.com/facebook/jest/releases)
- [Changelog](https://github.com/facebook/jest/blob/master/CHANGELOG.md)
- [Commits](https://github.com/facebook/jest/commits/v27.0.1/packages/jest-docblock)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump core-js from 3.12.1 to 3.13.1 (#10983)

Bumps [core-js](https://github.com/zloirock/core-js/tree/HEAD/packages/core-js) from 3.12.1 to 3.13.1.
- [Release notes](https://github.com/zloirock/core-js/releases)
- [Changelog](https://github.com/zloirock/core-js/blob/master/CHANGELOG.md)
- [Commits](https://github.com/zloirock/core-js/commits/v3.13.1/packages/core-js)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump ci-info from 3.1.1 to 3.2.0 (#10977)

Bumps [ci-info](https://github.com/watson/ci-info) from 3.1.1 to 3.2.0.
- [Release notes](https://github.com/watson/ci-info/releases)
- [Changelog](https://github.com/watson/ci-info/blob/master/CHANGELOG.md)
- [Commits](https://github.com/watson/ci-info/compare/v3.1.1...v3.2.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump jest-watch-typeahead from 0.6.3 to 0.6.4 (#10972)

Bumps [jest-watch-typeahead](https://github.com/jest-community/jest-watch-typeahead) from 0.6.3 to 0.6.4.
- [Release notes](https://github.com/jest-community/jest-watch-typeahead/releases)
- [Changelog](https://github.com/jest-community/jest-watch-typeahead/blob/main/CHANGELOG.md)
- [Commits](https://github.com/jest-community/jest-watch-typeahead/compare/v0.6.3...v0.6.4)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump eslint-plugin-import from 2.23.3 to 2.23.4 (#10969)

Bumps [eslint-plugin-import](https://github.com/benmosher/eslint-plugin-import) from 2.23.3 to 2.23.4.
- [Release notes](https://github.com/benmosher/eslint-plugin-import/releases)
- [Changelog](https://github.com/benmosher/eslint-plugin-import/blob/master/CHANGELOG.md)
- [Commits](https://github.com/benmosher/eslint-plugin-import/compare/v2.23.3...v2.23.4)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump @glimmer/syntax from 0.79.2 to 0.79.3 (#10986)

Bumps [@glimmer/syntax](https://github.com/glimmerjs/glimmer-vm) from 0.79.2 to 0.79.3.
- [Release notes](https://github.com/glimmerjs/glimmer-vm/releases)
- [Changelog](https://github.com/glimmerjs/glimmer-vm/blob/master/CHANGELOG.md)
- [Commits](https://github.com/glimmerjs/glimmer-vm/compare/v0.79.2...v0.79.3)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump flow-parser from 0.151.0 to 0.152.0 (#10984)

Bumps [flow-parser](https://github.com/facebook/flow) from 0.151.0 to 0.152.0.
- [Release notes](https://github.com/facebook/flow/releases)
- [Changelog](https://github.com/facebook/flow/blob/master/Changelog.md)
- [Commits](https://github.com/facebook/flow/compare/v0.151.0...v0.152.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.37.1 to 5.38.1 (#10981)

Bumps [webpack](https://github.com/webpack/webpack) from 5.37.1 to 5.38.1.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.37.1...v5.38.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump babel-jest from 27.0.1 to 27.0.2 (#10980)

Bumps [babel-jest](https://github.com/facebook/jest/tree/HEAD/packages/babel-jest) from 27.0.1 to 27.0.2.
- [Release notes](https://github.com/facebook/jest/releases)
- [Changelog](https://github.com/facebook/jest/blob/master/CHANGELOG.md)
- [Commits](https://github.com/facebook/jest/commits/v27.0.2/packages/babel-jest)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @babel/preset-env from 7.14.2 to 7.14.4 (#10978)

Bumps [@babel/preset-env](https://github.com/babel/babel/tree/HEAD/packages/babel-preset-env) from 7.14.2 to 7.14.4.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.14.4/packages/babel-preset-env)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump actions/cache from 2.1.5 to 2.1.6 (#10965)

Bumps [actions/cache](https://github.com/actions/cache) from 2.1.5 to 2.1.6.
- [Release notes](https://github.com/actions/cache/releases)
- [Commits](https://github.com/actions/cache/compare/v2.1.5...v2.1.6)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump eslint-plugin-regexp from 0.10.0 to 0.11.0 (#10966)

Bumps [eslint-plugin-regexp](https://github.com/ota-meshi/eslint-plugin-regexp) from 0.10.0 to 0.11.0.
- [Release notes](https://github.com/ota-meshi/eslint-plugin-regexp/releases)
- [Commits](https://github.com/ota-meshi/eslint-plugin-regexp/compare/v0.10.0...v0.11.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump jest from 27.0.1 to 27.0.3 (#10976)

sosukesuzuki committed:
Fix order of override modifiers (#10961)

* Fix modifiers order

* Add tests

* Add changelog

dependabot[bot] committed:
Build(deps): Bump @babel/parser from 7.14.3 to 7.14.4 (#10975)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @glimmer/reference from 0.79.2 to 0.79.3 (#10970)

Bumps @glimmer/reference from 0.79.2 to 0.79.3.

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump eslint-plugin-react from 7.23.2 to 7.24.0 (#10967)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump concurrently from 6.1.0 to 6.2.0 in /website (#10968)

Bumps [concurrently](https://github.com/kimmobrunfeldt/concurrently) from 6.1.0 to 6.2.0.
- [Release notes](https://github.com/kimmobrunfeldt/concurrently/releases)
- [Commits](https://github.com/kimmobrunfeldt/concurrently/compare/v6.1.0...v6.2.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump codemirror-graphql from 1.0.1 to 1.0.2 in /website (#10971)

Bumps [codemirror-graphql](https://github.com/graphql/graphiql) from 1.0.1 to 1.0.2.
- [Release notes](https://github.com/graphql/graphiql/releases)
- [Changelog](https://github.com/graphql/graphiql/blob/main/CHANGELOG.md)
- [Commits](https://github.com/graphql/graphiql/compare/graphiql@1.0.1...codemirror-graphql@1.0.2)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @babel/types from 7.14.2 to 7.14.4 (#10974)

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.37.1 to 5.38.1 in /website (#10973)

dependabot[bot] committed:
Build(deps): Bump angular-estree-parser from 2.3.0 to 2.4.0 (#10992)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump execa from 5.0.0 to 5.0.1 (#10993)

Bumps [execa](https://github.com/sindresorhus/execa) from 5.0.0 to 5.0.1.
- [Release notes](https://github.com/sindresorhus/execa/releases)
- [Commits](https://github.com/sindresorhus/execa/compare/v5.0.0...v5.0.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
test: use file: link for eslint-plugin-prettier-internal-rules

with some workaround hacks

to fix `npm clean dev test` action

Commit:
fixup! test: use file: link for eslint-plugin-prettier-internal-rules

dependabot[bot] committed:
Build(deps): Bump @angular/compiler from 11.2.14 to 12.0.2 (#10995)

Bumps [@angular/compiler](https://github.com/angular/angular/tree/HEAD/packages/compiler) from 11.2.14 to 12.0.2.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.2/packages/compiler)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

adalinesimonian committed:
fix(ci/tests): install optional deps when testing

adalinesimonian committed:
fix(lint): whoops, add missing semicolon

adalinesimonian committed:
fix(tests/env): prettier → prettierx

fisker committed:
Fix `StringLiteral` print (#10997)

dependabot[bot] committed:
Build(deps): Bump linguist-languages from 7.14.0 to 7.15.0 (#10990)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: fisker Cheung <lionkay@gmail.com>

adalinesimonian committed:
fix(js/modules): add missing separators to imports

adalinesimonian committed:
[XXX TODO] prettierx: set version to 0.19.0-dev

@adalinesimonian TODO: please do this before merging changes from Prettier 2.3.0

brody2consult commented on scripts/tools/eslint-plugin-prettier-internal-rules/no-node-comments.js:
I added this comment to avoid reformatting this entire return statement ... argh!

brody2consult commented on scripts/tools/eslint-plugin-prettier-internal-rules/test.js:
minor nit __OPTIONAL__:

I think it would be ideal if we could find a way to get these tests working with the updates I made to fix `npm clean dev test`.

brody2consult commented on src/document/doc-debug.js:
Considering that this is for debugging, I am wondering if we should just do what they did in Prettier 2.3.0?

adalinesimonian committed:
feat(test): disable tests by file, parser, options - REAPPLY THIS WORKAROUND

The babel-ts parser errors out on valid syntax in some cases. To
maximize coverage, instead of disabling tests being run on babel-ts for
an entire suite, this functionality allows disabling tests by filename,
parser, and options.

adalinesimonian committed:
feat(test): disable tests by file, parser, options

The babel-ts parser errors out on valid syntax in some cases. To
maximize coverage, instead of disabling tests being run on babel-ts for
an entire suite, this functionality allows disabling tests by filename,
parser, and options.

adalinesimonian committed:
Merge branch 'adalinesimonian/merge-prettier-2.3.0' of https://github.com/adalinesimonian/prettierx into adalinesimonian/merge-prettier-2.3.0

Commit:
Revert "feat(test): disable tests by file, parser, options"

This reverts commit d8a42e3539a5e5c2a5a92e04ad401775071ebfeb.

The workaround should not be needed with @babel/parser -> 7.14.4 update.

Commit:
prettierx: update @babel/parser -> 7.14.4

Commit:
FIX-UP: only skip disabled test in round-trip stage

adalinesimonian committed:
feat(test): allow disabling tests for round trips

- Allow determining whether or not to disable a test based on whether or
  not the test run is a round trip
- Added documentation comments

adalinesimonian commented on tests/config/format-test.js:
This change is good because it means we don't avoid testing the `babel-ts` tests when they don't run into the bug we discovered, but it also means that `isDisabled` is a little less clear. I imagine a scenario in which someone adds another test to the disabled test list, assuming that they will be disabled in their entirety, even though that isn't the behaviour.

I've pushed up a small change that, I think, makes this a little clearer without making any overly complicated additions or changes. I also threw in some doc comments while I was at it.

brody2consult commented on tests/config/format-test.js:
Thanks @adalinesimonian, but I don't think we need to disable any tests at this point in the module. I started refactoring to rename the function to `skipRoundTripErrorTest` and make the changes above a little more consistent with `unstableTests`.

Commit:
FIX-UP (further refactoring) re: disabling tests for round trips

Commit:
Revert "prettierx: update @babel/parser -> 7.14.4" - leaving this for another task

This reverts commit 427d816e93aea28deb45b5acd58ae237867d9a5a.

Commit:
FIX-UP: remove tslib from package dependencies, etc.

Commit:
FIX-UP: remove json-stable-stringify - now replaced by fast-json-stable-stringify

Commit:
add TEMPORARY EXCEPTION to prettierx dependencies notes (...)

Commit:
FIX-UP: fix CHANGELOG.md

aliaksandrsen committed:
Update v2.3 blog post (#11002)

Co-authored-by: fisker Cheung <lionkay@gmail.com>

adalinesimonian committed:
chore: use broken-out bracket spacing options

Commit:
FIX-UP: remove strip-ansi from devDependencies ... but strip-ansi MUST REMAIN in package dependencies

Commit:
FIX-UP: remove dedent & srcset NOT NEEDED from package dependencies

Commit:
FIX-UP: remove some devDependencies not needed

Commit:
FIX-UP: add missing [prettierx] comments to jest.config.js

Commit:
[FIX-UP] remove dedent from scripts/release/package.json for consistency with Prettier 2.3.0

Commit:
[FIX-UP] remove entries not needed from scripts/release/yarn.lock for consistency with Prettier 2.3.0

adalinesimonian committed:
Merge branch 'adalinesimonian/merge-prettier-2.3.0' of https://github.com/adalinesimonian/prettierx into adalinesimonian/merge-prettier-2.3.0

brody2consult commented on .editorconfig:
I think it would be ideal to propose these changes in the upstream Prettier project.

brody2consult commented on cspell.json:
This should be removed from upstream Prettier, you can leave this one up to me.

brody2consult commented on src/language-css/options.js:
ditto: I think it would be ideal to propose changes like this in the upstream Prettier project.

adalinesimonian committed:
chore: remove option inheritance

Will be moved to a separate PR

adalinesimonian commented on tests/config/format-test.js:
Agreed, better to keep things simple.

adalinesimonian commented on .editorconfig:
Agreed - these changes aren't specific to prettierX.

brody2consult commented on tests/format/js/bracket-spacing/jsfmt.spec.js:
Do we need *all* of these broken-out spacing options here?

brody2consult commented on tests/format/js/export/jsfmt.spec.js:
ditto: Do we need *all* of these broken-out spacing options here?

brody2consult commented on tests/format/js/import-assertions/bracket-spacing/jsfmt.spec.js:
ditto (Do we need *all* of these broken-out spacing options here?)

brody2consult commented on website/playground/Playground.js:
Interesting, did you try building this playground?

Commit:
FIX-UP: update tests/format/angular/angular/jsfmt.spec.js to improve consistency with Prettier 2.3.0

Commit:
[FIX-UP] update tests/format/angular/angular/__snapshots__/jsfmt.spec.js.snap

adalinesimonian committed:
test(bracket-spacing): only set necessary options

adalinesimonian committed:
Merge branch 'adalinesimonian/merge-prettier-2.3.0' of https://github.com/adalinesimonian/prettierx into adalinesimonian/merge-prettier-2.3.0

adalinesimonian commented on tests/format/js/bracket-spacing/jsfmt.spec.js:
Whoops, forgot to whittle these down. Just pushed an update that keeps only necessary options.

adalinesimonian commented on tests/format/js/export/jsfmt.spec.js:
ditto: (Whoops, forgot to whittle these down. Just pushed an update that keeps only necessary options.)

adalinesimonian commented on tests/format/js/import-assertions/bracket-spacing/jsfmt.spec.js:
ditto: (Whoops, forgot to whittle these down. Just pushed an update that keeps only necessary options.)

Commit:
[FIX-UP] update tests/format/yaml/flow-mapping/jsfmt.spec.js to improve consistenty with snapshot from Prettier 2.3.0

Commit:
[FIX-UP] update tests/format/yaml/flow-mapping/__snapshots__/jsfmt.spec.js.snap

Commit:
[FIX-UP] update tests/format/yaml/flow-sequence/jsfmt.spec.js to improve consistenty of snapshot with Prettier 2.3.0

Commit:
[FIX-UP] update tests/format/yaml/flow-sequence/__snapshots__/jsfmt.spec.js.snap

Commit:
prettierx: add another objectCurlySpacing option comment

Commit:
Merge branch 'dev' of https://github.com/brodybits/prettierx into adalinesimonian/merge-prettier-2.3.0

Commit:
FIX-UP (breakBeforeElse): add prettierx comment

Commit:
MERGE FIX-UP: keep src/document/doc-debug.js from Prettier 2.3.0

brody2consult commented on src/language-js/needs-parens.js:
🎉 Yeah !!

I *may* update these comments a little and do some more testing.

adalinesimonian commented on src/language-js/needs-parens.js:
> I _may_ update these comments a little and do some more testing.

Go for it! Before I commented the additional logic out, it was causing all sorts of problems all over the place, getting rid of parentheses in cases where they shouldn't have been; one specific example I remember being

```js
!(
  condition1 &&
  condition2
)
```

becoming

```js
!
  condition1 &&
  condition2

```

Commit:
FIX-UP: remove "PrettierX" entries not needed from cspell.json

Commit:
[TBD ???] move the new prettierX cspell entries near the end of the words list

FUTURE TODO I would like to move the other prettierX-specific cspell entries near the end as well
(likely outside the merge-prettier-2.3.0 PR)

Commit:
FIX-UP: remove inherit property type no longer needed from src/main/core-options.js

Commit:
FIX-UP: REVERT my updates to cspell.json - my bad, moved into a separate PR

---

Revert "[TBD ???] move the new prettierX cspell entries near the end of the words list"

This reverts commit 225459041adbaa0822e05b6d15cb616e3aecf2c8.

---

Revert "FIX-UP: remove "PrettierX" entries not needed from cspell.json"

This reverts commit 9510335c4202bcb8ff90c1d9a6690a16e7dcd1c7.

Commit:
MERGE FIX-UP: add & update some comments for prettierX in src/language-js

adalinesimonian commented on website/playground/Playground.js:
I didn't - but then I did to see if it'd work. With a few small changes it did:

![image](https://user-images.githubusercontent.com/9868643/120565004-febf0200-c3c0-11eb-8669-fcaa87ba786c.png)

However, I think getting the playground/site to work may be beyond the scope of this PR. On the one hand, there are changes to the site upstream. But I'm not certain that's sufficient cause. Should we omit the changes to the site entirely, then, from this PR, and separate them into their own?

Commit:
MERGE FIX-UP: add prettierx-specific package files entries

renovate[bot] committed:
prettierx: Update get-stream -> 6.0.1 - dependency (#528)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
prettierx: Update mem -> 8.1.1 - dependency (#537)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
prettierx: Update chalk -> 4.1.1 - dependency (#538)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
prettierx: Update eslint-plugin-prettier -> 3.4.0 - devDependency (#525)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
prettierx: Update eslint -> 7.27.0 - devDependency (#540)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
prettierx: Update jest-watch-typeahead -> 0.6.4 - devDependency (#539)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

Commit:
prettierx: version 0.18.1

Commit:
prettierx: start 0.19.0-dev

adalinesimonian committed:
chore: remove changes that should be made upstream

adalinesimonian committed:
Merge branch 'adalinesimonian/merge-prettier-2.3.0' of https://github.com/adalinesimonian/prettierx into adalinesimonian/merge-prettier-2.3.0

adalinesimonian committed:
chore: build → build-extra-dist, will separate PR

Commit:
prettierx: remove GitHub items not wanted from Prettier 2.3.0

- .github/workflows/mark-issue-duplicate.yml
- .github/workflows/support.yml

Commit:
prettierx: remove GitHub items not wanted from Prettier 2.3.0

- .github/workflows/lock.yml
- .github/workflows/mark-issue-duplicate.yml
- .github/workflows/support.yml

adalinesimonian committed:
Merge branch 'adalinesimonian/merge-prettier-2.3.0' of https://github.com/adalinesimonian/prettierx into adalinesimonian/merge-prettier-2.3.0

adalinesimonian committed:
fix(lint): allow npm install w/ no-node-comments

brody2consult commented on scripts/tools/eslint-plugin-prettier-internal-rules/no-node-comments.js:
Shouldn't this be fixed in upstream Prettier?

Otherwise ... yeah!!

adalinesimonian commented on scripts/tools/eslint-plugin-prettier-internal-rules/no-node-comments.js:
Yeah, it definitely belongs upstream. I left it in while testing so that I could tell when yarn/npm switched to using the updated copy, but I forgot to pull it out before committing.

adalinesimonian committed:
chore: revert doc url edit meant for upstream

Commit:
MERGE FIX-UP: use Babel by default for Flow in src/language-js/index.js

Commit:
MERGE FIX-UP: update tests/integration/__tests__/__snapshots__/support-info.js.snap

Commit:
Merge branch 'dev' of https://github.com/brodybits/prettierx into adalinesimonian/merge-prettier-2.3.0

Commit:
FIX-UP: .github items

Commit:
FIX-UP CHANGELOG.md - my bad

Commit:
FIX-UP: add details from Prettier 2.3.0 to CONTRIBUTING.md etc.

Commit:
[MERGE FIX-UP] improve section headers in CONTRIBUTING.md

adalinesimonian committed:
Fix docs urls for internal lint rules (#11011)

scripts/eslint-plugin-prettier-internal-rules →
scripts/tools/eslint-plugin-prettier-internal-rules

Commit:
Fix a typo (#11009)

Need to change 'M' to 'm' otherwise it won't work. You may refer to https://github.com/neoclide/coc-prettier/pull/47

fisker committed:
Remove `@types/node` package (#10994)

dcyriller committed:
Preserve attributes order for element node (#10958)

* Preserve attributes order for element node

* Add CHANGELOG

* Add ...arguments in the test

* Apply fisker suggestions

* Simplify further

sosukesuzuki committed:
Revert features for minor (#11016)

This reverts commit f8cb7857587527267c4cff06631415ced3f06a89.

sosukesuzuki committed:
Release 2.3.1

adalinesimonian committed:
fix(format): restore --no-indent-chains support

Commit:
[NIT FIX-UP] add comment in place of code that was moved & updated

adalinesimonian committed:
Merge tag '2.3.1' into adalinesimonian/merge-prettier-2.3.0

Prettier 2.3.1

adalinesimonian committed:
fix(build): util → utils

adalinesimonian committed:
fix(support): update prettier version

adalinesimonian commented on src/main/support.js:
I think this would be much less likely to get overlooked if we moved it to package.json, for example:

```js
const currentVersion = require("../../package.json")["prettier-version"];
```

```json
{
  "name": "prettierx",
  "version": "0.19.0-dev",
  "prettier-version": "2.3.1",
  "description": "prettierx - less opinionated code formatter fork of prettier",
  "bin": "./bin/prettierx.js",
  "repository": "brodybits/prettierx",
```

brody2consult commented on src/main/support.js:
👍
